Sources/AWSXRayInstrument/Span.swift

```diff
@@ -0,0 +1,72 @@
+import AWSXRayRecorder
+import Baggage
+import Dispatch
+import Instrumentation
+
+extension XRayRecorder.Segment: Instrumentation.Span {
+    public var operationName: String {
+        // TODO: impl
+        "test"
+    }
+
+    public var kind: SpanKind {
+        // TODO: impl
+        .internal
+    }
+
+    public var status: SpanStatus? {
+        // TODO: impl
+        get {
+            nil
+        }
+        set(newValue) {}
+    }
+
+    public var startTimestamp: DispatchTime {
+        // TODO: impl
+        DispatchTime.now()
+    }
+
+    public var endTimestamp: DispatchTime? {
+        // TODO: impl
+        nil
+    }
+
+    public func end(at timestamp: DispatchTime) {
+        // TODO: impl
+    }
+
+    public var baggage: BaggageContext {
+        // TODO: impl
+        BaggageContext()
+    }
+
+    public var events: [SpanEvent] {
+        // TODO: impl
+        [SpanEvent]()
+    }
+
+    public func addEvent(_: SpanEvent) {
+        // TODO: impl
+    }
+
+    public var attributes: SpanAttributes {
```

Comment:
While poking around I bridged `Annotations` <-> `SpanAttributes` like this. Note that I had access to `_annotations` as I added this to the `Segment` file directly. If the `TracingInstrument` and `Span` conformance ends up in a separate library there'd have to be a way to access the annotations. One way would be to create a `forEach` similar to what we have [in the `SpanAttributes` type](https://github.com/slashmo/gsoc-swift-tracing/blob/main/Sources/Instrumentation/Tracing/Span.swift#L193).

```swift
public var attributes: SpanAttributes {
    get {
        SpanAttributes(self._annotations.mapValues(SpanAttribute.init))
    }
    set(attributes) {
        attributes.forEach { key, attribute in
            self.setAnnotation(key, attribute: attribute)
        }
    }
}

extension SpanAttribute {
    init(_ annotationValue: XRayRecorder.Segment.AnnotationValue) {
        switch annotationValue {
        case let .string(string):
            self = .string(string)
        case let .bool(bool):
            self = .bool(bool)
        case let .float(float):
            self = .double(Double(float))
        case let .int(int):
            self = .int(int)
        }
    }
}

extension XRayRecorder.Segment {
    func setAnnotation(_ key: String, attribute: SpanAttribute) {
        switch attribute {
        case let .string(string):
            self.setAnnotation(key, value: string)
        case let .int(int):
            self.setAnnotation(key, value: int)
        case let .double(double):
            self.setAnnotation(key, value: Float(double))
        case let .bool(bool):
            self.setAnnotation(key, value: bool)
        case let .stringConvertible(stringConvertible):
            self.setAnnotation(key, value: String(describing: stringConvertible))
        case .array:
            break
        }
    }
}
```

Reply:
Does the bridging here make sense with XRay's semantics (I don't know :-))?

And extra question, if it does, the current xray seems to only define `float` but not `double`s is this on purpose following some xray roles or an omission?

It'd cause this potential trouble: `           self.setAnnotation(key, value: Float(double))` when invoked through the generic API.

Reply:
> And extra question, if it does, the current xray seems to only define float but not doubles is this on purpose following some xray roles or an omission?
> It'd cause this potential trouble: self.setAnnotation(key, value: Float(double)) when invoked through the generic API.

that's a omission, also please check https://github.com/slashmo/gsoc-swift-tracing/issues/83